lib/compiler.js

```diff
@@ -76,12 +76,16 @@ function sanitize(config) {
     delete result.reporter;
     delete result.watchStdin;
 
-    // TODO: remove the need for this
-    for (const property of ['entry', 'output']) {
-      const target = result[property];
-      if (target && Object.keys(target).length === 0) {
-        delete result[property];
-      }
+    // TODO: remove the need for these cleanups
+    if (
+      result.entry &&
+      Object.keys(result.entry).length === 0 &&
+      !(result.entry instanceof Function)
```

Comment:
The `instanceof` check here is flawed:

``` console
$ const p = new Promise((r, x)=>{})
$ p instanceof Function
false
$ typeof p
'object'
$ typeof p.then
'function'
$ const f = () => {}
$ typeof f
'function'
```

Reply:
Neither the docs nor webpack allow entry to be PromiseLike (and changing that would be a breaking for anyone with a 'then' entrypoint).

The case you're likely thinking of is that entry may be a function that _returns_ PromiseLike.